### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by EatenCameraman**
JunHyuk Kim 
First Task ISIC 2017 Improved Unet
Student Number 47423202

### CODE REVIEW

**Comment by EatenCameraman** [comment]
JunHyuk Kim
First Task ISIC 2017 Improved Unet
Student Number 47423202

Pull request made before finishing the task.

**Comment by fallschirmfaust** [comment]
JunHyuk Kim
First Task ISIC 2017 Improved Unet
Student Number 47423202

**Comment by fallschirmfaust** [comment]
JunHyuk Kim
First Task ISIC 2017 Improved Unet
Student Number 47423202
Final Edit completed the task.

**Comment by fallschirmfaust** [comment]
JunHyuk Kim
First Task ISIC 2017 Improved Unet
Student Number 47423202
Final Edit completed the task.

**Comment by fallschirmfaust** [comment]
JunHyuk Kim
First Task ISIC 2017 Improved Unet
Student Number 47423202
Final Edit completed the task.

**Comment by EatenCameraman** [comment]
JunHyuk Kim
First Task ISIC 2017 Improved Unet
Student Number 47423202
Final Edit Submitted to Turnitin.

**Comment by EatenCameraman** [comment]
JunHyuk Kim
First Task ISIC 2017 Improved Unet
Student Number 47423202
Final confirmation submission. Removed a bit of commentation and Bit of readme.

**Comment by EatenCameraman** [comment]
JunHyuk Kim
First Task ISIC 2017 Improved Unet
Student Number 47423202
DIdnt realise the branch was closed. Reopened it

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Wrong files path. All student constructed document should be packed in one document under the ‘recognition’ document.
- Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.8383. Predicted Mask output looks good.
- Try to use PyTorch to construct the Improved U-Net model, but the model body is roughly a U-Net-like model(filters start with 64 and end up with 512) instead of the improved U-Net one(filters start with 16 and end up with 256). Other parts are ok.
- Good coding design.
- Sufficient docstring and comment in the code.
- Easy difficulty problem, no higher difficulty marks.

**Comment by EatenCameraman** [comment]
About the wrong file path, you mean I have two folders and one should be removed? Or is there more? 

Secondly, is it required for me to use tensorflow? I dont think Ive used tensorflow before, so Im not sure if I can do that in few days while also preparing for my exams. I can try, but Im a bit worried.

**Comment by fallschirmfaust** [comment]
Lowering it from 64 to 16 actually lowers the accuracy. Its still around 80, is this expected?

**Comment by wangzhaomxy** [comment]
> About the wrong file path, you mean I have two folders and one should be removed? Or is there more?
> 
> Secondly, is it required for me to use tensorflow? I dont think Ive used tensorflow before, so Im not sure if I can do that in few days while also preparing for my exams. I can try, but Im a bit worried.

This is just comments for marking. No need to change anything.
Sorry for the TensorFlow comment. It's a typo. I have righted the wrong.
For the wrong file path, you should put in only one document under the topic-recognition, as all students files will be merged in this document.

**Comment by EatenCameraman** [comment]
Does the change look good to you? 
I'm not sure how to apply the feedback. I changed a bit, because I need to apply the feedback within a week right? 
Does this look good?

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
OK design and implementation but incorrect model (UNet) -4
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove pyc files for merging -2
Request Description minimal -1

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by EatenCameraman** [comment]
I changed the file structure and removed the pyc files. Do you see this and is this good enough for the feedback mark?

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2.

**Comment by EatenCameraman** [comment]
Should I close the branch?  Or should I keep this open?
